src/pip/_internal/cli/cmdoptions.py

```diff
@@ -1004,7 +1004,7 @@ def check_list_path_option(options: Values) -> None:
     choices=[
         "fast-deps",
         "truststore",
-        "no-binary-enable-wheel-cache",
+        "no-binary-enable-wheel-cache",  # now always on
```

Comment:
Why not remove this, as we've done with other `--use-feature` flags?

Reply:
I kind of remember folks complaining that we remove feature flags that became always on because it was causing unnecessary churn and I was kind of agreeing. But yeah if we keep it we probably need some warning that it's going away one day. Not sure what's best.

Reply:
Let's leave it in with a warning -- it's less churn for end users. However, I do feel like we should print these warnings in a single central location, instead of scattering them across the commands. Do that seem reasonable?

If so, the only place that I can think of at the moment is `Command._main`, but I trust your judgement on determining a good place.

Reply:
@pradyunsg I have centralized the handling of always enabled features, so it will be easy to use for future similar cases.
I do that immediately after setting up the logger so we get the colored warning.